utils/python/CIME/case_run.py

```diff
@@ -89,6 +89,11 @@ def run_model(case):
     cmd = case.get_resolved_value(cmd)
     logger.info("run command is %s " %cmd)
 
+    with open("runtime_environment.txt", 'w') as fd:
```

Comment:
This should probably follow the same process as case_setup for recording the environment:
```
        with open("software_environment.txt", "w") as f:
            f.write(env_module.list_modules())
        run_cmd_no_fail("echo -e '\n' >> software_environment.txt && env >> software_environment.txt")
```
This might be worth encapsulating in env_mach_specific since we are now doing it in two places.